cpp/tests/test_compartments_simulation.cpp

```diff
@@ -55,6 +56,38 @@ TEST(TestCompartmentSimulation, integrator_uses_model_reference)
     ASSERT_NEAR(sim.get_result().get_last_value()[0], 3.0, 1e-5);
 }
 
+TEST(TestCompartmentSimulation, copy_simulation)
+{
+
+    auto sim = mio::Simulation<double, MockModel>(MockModel(), 0.0);
+
+    auto sim_copy_cnstr = mio::Simulation<double, MockModel>(sim);
+    auto sim_copy_assign = sim;
+
+    EXPECT_EQ(sim.get_model().m_dydt, sim_copy_cnstr.get_model().m_dydt);
+    EXPECT_EQ(sim.get_model().m_dydt, sim_copy_assign.get_model().m_dydt);
+
+    // modifying model should not effect copies
+    sim.get_model().m_dydt = 2.0;
+    sim.advance(1.0);
+    sim.get_integrator().get_dt_max() = 2.0; 
+
+    EXPECT_NE(sim_copy_cnstr.get_model().m_dydt, 2.0);
+    EXPECT_NE(sim_copy_assign.get_model().m_dydt, 2.0);
+    EXPECT_NE(sim_copy_cnstr.get_integrator().get_dt_max(), 2.0);
+    EXPECT_NE(sim_copy_assign.get_integrator().get_dt_max(), 2.0);
+    
+    // copy simulation after advance
+    sim_copy_cnstr = mio::Simulation<double, MockModel>(sim);
+    sim_copy_assign = sim;
+
+    EXPECT_EQ(sim.get_result().get_last_value()[0], sim_copy_cnstr.get_result().get_last_value()[0]);
+    EXPECT_EQ(sim.get_result().get_last_value()[0], sim_copy_assign.get_result().get_last_value()[0]);
+    EXPECT_EQ(sim_copy_cnstr.get_integrator().get_dt_max(), 2.0);
+    EXPECT_EQ(sim_copy_assign.get_integrator().get_dt_max(), 2.0);
```

Comment:
I think here it would be more interesting to call advance on the copies as well, then check that all three are in the same state again.

For that, you could put the modification (lines 71-73) in a lambda taking a sim as argument.